### PR TITLE
Add support for base spacing scale in Stack

### PR DESCRIPTION
**Pull request by danielguillan**
## Summary

This pull request adds support for [base spacing scale tokens](https://github.com/primer/primitives/blob/main/src/tokens/base/size/size.json) on Stack's `gap` and `padding` props. This allows for more flexible customization of spacing using Stack while keeping the possible values constrained to our size tokens.

Basic example:

```
<Stack gap={96} padding={40}>...</Stack>
```

With responsive props

```
<Stack
  gap={{
    narrow: 'condensed',
    regular: 48,
    wide: 96, 
  }}
  padding={40}
>
 ...
</Stack>
```

- [Docs](https://primer-40747357d8-26139705.drafts.github.io/components/Stack)

## List of notable changes:

- Extended `StackSpacingVariants` to include size scale values
- Added classes for gap and padding to support the new scale
- Added test coverage for Stack

## Supporting resources (related issues, external links, etc):

- Fixes https://github.com/github/primer/issues/2237

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### CODE REVIEW

File: packages/react/src/Stack/Stack.test.tsx [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    expect(getByText(1)).toBeInTheDocument()
+    expect(getByText('Fragment')).toBeInTheDocument()
+  })
+
+  it('should set the correct styles for direction', () => {
+    const mockTestId = 'test'
+    const expectedClass = 'Stack--vertical'
+
+    const {getByTestId} = render(
+      <Stack direction={'vertical'} data-testid={mockTestId}>
```
Comment (rezrah) [inline packages/react/src/Stack/Stack.test.tsx:44]: Nitpicking, sorry... can we stick to string types as there's no interpolation?

```suggestion
      <Stack direction="vertical" data-testid={mockTestId}>
```

File: packages/react/src/constants.tsx [hunk 2]
```diff
@@ -16,3 +16,6 @@ export const Colors = [
 ] as const
 
 export const Gradients = ['blue-purple', 'green-blue', 'pink-blue', 'purple-red', 'red-orange'] as const
+
+export const BaseSizeScale = [4, 8, 12, 16, 20, 24, 28, 32, 36, 40, 44, 48, 64, 80, 96, 112, 128] as const
```
Comment (rezrah) [inline packages/react/src/constants.tsx:20]: Love this. I think we should generate this using style dictionary and serve from our brand-primitives package in the long term, as it's tightly coupled to a dependency. Could we add a FIXME or TODO above this line to indicate this?

File: .changeset/tame-pigs-appear.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@primer/react-brand': patch
+---
+
+Adds spacing scale support in Stack's gap and padding props
```
Comment (rezrah) [inline .changeset/tame-pigs-appear.md:5]: Is it possible to add some code examples for the changelog please?

Something like:

```jsx
<>
<Stack gap={96} padding={96}>...</Stack>
<Stack 
  gap={{
    narrow: 96
  }} 
  padding={{
    narrow: 96
   }}>
  ...
</Stack>
</>
```

**Review by rezrah — COMMENTED** [review]
Looks great @danielguillan. Could we also update the [docs please](https://primer.style/brand/components/Stack/)? And a storybook story would be great to increase our visual test coverage.

**Comment by danielguillan** [comment]
@rezrah thanks! I applied the feedback and added a new story for the base scale support.

**Review by josepmartins — APPROVED** [review]
LGTM! ✨

**Review by rezrah — APPROVED** [review]
This is such a huge feature. Amazing work @danielguillan 👏.

**Pull request merged**